### PR TITLE
Split out the WebSocket client

### PR DESCRIPTION
**Pull request by stuartwdouglas**
This allows the client to be used without
pulling in vertx-http.

### CODE REVIEW

File: bom/application/pom.xml [hunk 1]
```diff
@@ -1612,6 +1612,16 @@
                 <artifactId>quarkus-websockets-deployment</artifactId>
                 <version>${project.version}</version>
             </dependency>
+            <dependency>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-websockets-client</artifactId>
```
Comment (cescoffier) [inline bom/application/pom.xml:1617]: why websockets (plural) for the client and websocket (singular) for the server? 
I would go with singular everywhere, but maybe I'm missing the reason.

File: core/deployment/src/main/java/io/quarkus/deployment/Feature.java [hunk 2]
```diff
@@ -142,6 +142,7 @@
     SWAGGER_UI,
     TIKA,
     WEBSOCKETS,
+    WEBSOCKETS_CLIENT,
```
Comment (cescoffier) [inline core/deployment/src/main/java/io/quarkus/deployment/Feature.java:145]: So here, both are plural.

File: docs/src/main/asciidoc/websockets.adoc [hunk 3]
```diff
@@ -166,7 +167,8 @@ You can also test your web socket applications using the approach detailed {quic
 
 == WebSocket Clients
 
-Quarkus also contains a WebSocket client. You can call `ContainerProvider.getWebSocketContainer().connectToServer` to create a websocket connection.
+Quarkus also contains a WebSocket client. You can call `ContainerProvider.getWebSocketContainer().connectToServer` to create a websocket connection. By default the `quarkus-websockets`
+artifact includes both client and server support however if you only want the client you can include `quarkus-websockets-client` instead.
```
Comment (cescoffier) [inline docs/src/main/asciidoc/websockets.adoc:171]: Minor - there is a new line that is not required (wrapping probably).

File: extensions/websockets/client/deployment/src/test/java/io/quarkus/undertow/websockets/test/WebSocketClientTestCase.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            public void handle(ServerWebSocket serverWebSocket) {
+                serverWebSocket.writeTextMessage("Hello World").onComplete(new Handler<AsyncResult<Void>>() {
+                    @Override
+                    public void handle(AsyncResult<Void> voidAsyncResult) {
+                        serverWebSocket.close();
+                    }
+                });
+            }
+        });
+        server.listen(8081);
```
Comment (cescoffier) [inline extensions/websockets/client/deployment/src/test/java/io/quarkus/undertow/websockets/test/WebSocketClientTestCase.java:53]: you may want to block until the server is started.

File: extensions/websockets/client/deployment/src/test/java/io/quarkus/undertow/websockets/test/WebSocketClientTestCase.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    @AfterAll
+    public static void stop() {
+        if (server != null) {
+            server.close().onComplete(new Handler<AsyncResult<Void>>() {
+                @Override
+                public void handle(AsyncResult<Void> voidAsyncResult) {
+                    vertx.close();
```
Comment (cescoffier) [inline extensions/websockets/client/deployment/src/test/java/io/quarkus/undertow/websockets/test/WebSocketClientTestCase.java:62]: You may want to wait until the instance is closed.
Without you can have leaks and reach the max file descriptor limit.

File: extensions/websockets/client/runtime/src/main/resources/META-INF/quarkus-extension.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+name: "WebSockets Client"
+artifact: ${project.groupId}:${project.artifactId}:${project.version}
+metadata:
+  short-name: "websockets-client"
+  keywords:
+  - "websocket"
+  - "websockets"
+  - "web-socket"
+  - "web-sockets"
```
Comment (cescoffier) [inline extensions/websockets/client/runtime/src/main/resources/META-INF/quarkus-extension.yaml:9]: Never see this variant... But ok.

**Review by cescoffier — CHANGES_REQUESTED** [review]
The main question is whether WebSocket should be singular or plural.

File: core/deployment/src/main/java/io/quarkus/deployment/Feature.java (hunk 2 above)
Comment (stuartwdouglas) [reply core/deployment/src/main/java/io/quarkus/deployment/Feature.java:145]: The Quarkus artifacts are all plural.

File: extensions/websockets/client/runtime/src/main/resources/META-INF/quarkus-extension.yaml (hunk 6 above)
Comment (stuartwdouglas) [reply extensions/websockets/client/runtime/src/main/resources/META-INF/quarkus-extension.yaml:9]: This was copied from the existing one.

**Review by cescoffier — APPROVED** [review]

**Pull request merged**